gpkit/constraints/sgp.py

```diff
@@ -108,6 +105,8 @@ def localsolve(self, solver=None, verbosity=1, x0=None, reltol=1e-4,
             print("Beginning signomial solve.")
         self.gps = []  # NOTE: SIDE EFFECTS
         self.results = []
+        if x0 and mutategp:
+            self._gp = self.init_gp(self.substitutions, verbosity, x0)
```

Comment:
This only gets called in localsolve if an `x0` is manually specified. The reason that `self.init_gp` is in `__init__` is that we want to verify that this isn't a GP, and doing so any other way requires duplicate computation.